sdk/go/client.go

```diff
@@ -1,93 +1,124 @@
+	tls "crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	ioutil "io/ioutil"
+	http "net/http"
+	"os"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/iancoleman/orderedmap"
 )
 
 type Client struct {
-    modifyLock              sync.RWMutex
-    addr                    string
-    apikey                  string
-    appid                   string
-    action                  string
-    resquest                *orderedmap.OrderedMap
+	modifyLock sync.RWMutex
+	addr       string
+	apikey     string
+	appid      string
 }
 
 // For production environment, the web server should request a formally issued
 // certificate, and change the below skipVerify to False.
 const skipVerify = true
 
 func NewClient() (*Client, error) {
-    c := new(Client)
-    c.modifyLock.Lock()
-    defer c.modifyLock.Unlock()
-
-    c.addr = os.Getenv("EHSM_ADDR")
-    if c.addr == "" {
-        return nil, fmt.Errorf("Please export EHSM_ADDR.")
-    }
-    c.appid = os.Getenv("EHSM_APPID")
-    if c.appid == "" {
-        return nil, fmt.Errorf("Please export EHSM_APPID.")
-    }
-    c.apikey = os.Getenv("EHSM_APIKEY")
-    if c.apikey == "" {
-        return nil, fmt.Errorf("Please export EHSM_APIKEY.")
-    }
-
-    return c, nil
+	c := new(Client)
+	c.modifyLock.Lock()
+	defer c.modifyLock.Unlock()
+
+	c.addr = os.Getenv("EHSM_ADDR")
+	if c.addr == "" {
+		return nil, fmt.Errorf("Please export EHSM_ADDR.")
+	}
+	c.appid = os.Getenv("EHSM_APPID")
+	if c.appid == "" {
+		return nil, fmt.Errorf("Please export EHSM_APPID.")
+	}
+	c.apikey = os.Getenv("EHSM_APIKEY")
+	if c.apikey == "" {
+		return nil, fmt.Errorf("Please export EHSM_APIKEY.")
+	}
+
+	return c, nil
 }
 
-func (c *Client) ehsmHttpAction() (*http.Response, error) {
-    // create an insecure Transport
-    tr := &http.Transport{
-        TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
-    }
+func ehsmHttpAction(resquest *orderedmap.OrderedMap, url, action string) (map[string]interface{}, error) {
+	// create an insecure Transport
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
 
-    // create an insecure Client
-    client := &http.Client{Transport: tr}
+	// create an insecure Client
+	client := &http.Client{Transport: tr}
 
-    requestBody, err := json.Marshal(c.resquest)
-    
-    // call ehsm kms
-    resp, err := client.Post(c.addr + "/ehsm?Action=" + c.action, "application/json", bytes.NewBuffer(requestBody))
+	requestBody, err := json.Marshal(resquest)
 
-    return resp, err
+	// call ehsm kms
+	resp, err := client.Post(url+"/ehsm?Action="+action, "application/json", bytes.NewBuffer(requestBody))
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("ReadAll error:", err)
+		return nil, err
+	}
+
+	var respJSON map[string]interface{}
+	err = json.Unmarshal(body, &respJSON)
+	if err != nil {
+		fmt.Println("Unmarshal error:", err)
+		return nil, err
+	}
+	return respJSON, err
```

Comment:
why not check the json[code] here?

if resp["Code"] != 200 {
		return "", fmt.Errorf(resp["Message"].(string))
	}

Reply:
fix